2023-09-18 17:45:19.034133: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'cudart64_100.dll'; dlerror: cudart64_100.dll not found
2023-09-18 17:45:19.034326: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
INFO flwr 2023-09-18 17:45:26,598 | app.py:151 | Starting Flower server, config: ServerConfig(num_rounds=10, round_timeout=None)
INFO flwr 2023-09-18 17:45:26,684 | app.py:172 | Flower ECE: gRPC server running (10 rounds), SSL is disabled
INFO flwr 2023-09-18 17:45:26,684 | server.py:86 | Initializing global parameters
INFO flwr 2023-09-18 17:45:26,684 | server.py:273 | Requesting initial parameters from one random client
INFO flwr 2023-09-18 17:45:26,710 | server.py:277 | Received initial parameters from one random client
INFO flwr 2023-09-18 17:45:26,711 | server.py:88 | Evaluating initial parameters
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:45:26,727 | server.py:95 | initial parameters (loss, other metrics): 0.6931471805599453, {'accuracy': 0.44333333333333336}
INFO flwr 2023-09-18 17:45:26,727 | server.py:101 | FL starting
DEBUG flwr 2023-09-18 17:45:26,894 | server.py:223 | fit_round 1: strategy sampled 2 clients (out of 3)
DEBUG flwr 2023-09-18 17:45:26,926 | server.py:237 | fit_round 1 received 2 results and 0 failures
WARNING flwr 2023-09-18 17:45:26,928 | fedavg.py:243 | No fit_metrics_aggregation_fn provided
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:45:26,936 | server.py:125 | fit progress: (1, 0.4772083827409828, {'accuracy': 0.7733333333333333}, 0.20855200000000096)
DEBUG flwr 2023-09-18 17:45:26,936 | server.py:173 | evaluate_round 1: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 17:45:26,966 | server.py:187 | evaluate_round 1 received 4 results and 0 failures
WARNING flwr 2023-09-18 17:45:26,966 | fedavg.py:274 | No evaluate_metrics_aggregation_fn provided
DEBUG flwr 2023-09-18 17:45:26,966 | server.py:223 | fit_round 2: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 17:45:26,994 | server.py:237 | fit_round 2 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:45:27,005 | server.py:125 | fit progress: (2, 0.47058630843542043, {'accuracy': 0.77}, 0.2787482000000008)
DEBUG flwr 2023-09-18 17:45:27,006 | server.py:173 | evaluate_round 2: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 17:45:27,027 | server.py:187 | evaluate_round 2 received 4 results and 0 failures
DEBUG flwr 2023-09-18 17:45:27,027 | server.py:223 | fit_round 3: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 17:45:27,057 | server.py:237 | fit_round 3 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:45:27,069 | server.py:125 | fit progress: (3, 0.47057446153972515, {'accuracy': 0.77}, 0.34192720000000065)
DEBUG flwr 2023-09-18 17:45:27,069 | server.py:173 | evaluate_round 3: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 17:45:27,093 | server.py:187 | evaluate_round 3 received 4 results and 0 failures
DEBUG flwr 2023-09-18 17:45:27,093 | server.py:223 | fit_round 4: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 17:45:27,125 | server.py:237 | fit_round 4 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:45:27,136 | server.py:125 | fit progress: (4, 0.4705744972100784, {'accuracy': 0.77}, 0.40850589999999976)
DEBUG flwr 2023-09-18 17:45:27,136 | server.py:173 | evaluate_round 4: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:45:27,160 | server.py:187 | evaluate_round 4 received 5 results and 0 failures
DEBUG flwr 2023-09-18 17:45:27,161 | server.py:223 | fit_round 5: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:45:27,192 | server.py:237 | fit_round 5 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:45:27,204 | server.py:125 | fit progress: (5, 0.46989038504356073, {'accuracy': 0.7733333333333333}, 0.47654709999999945)
DEBUG flwr 2023-09-18 17:45:27,204 | server.py:173 | evaluate_round 5: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:45:27,226 | server.py:187 | evaluate_round 5 received 5 results and 0 failures
DEBUG flwr 2023-09-18 17:45:27,227 | server.py:223 | fit_round 6: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:45:27,257 | server.py:237 | fit_round 6 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:45:27,270 | server.py:125 | fit progress: (6, 0.46989277613899866, {'accuracy': 0.7733333333333333}, 0.5428340000000009)
DEBUG flwr 2023-09-18 17:45:27,271 | server.py:173 | evaluate_round 6: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:45:27,294 | server.py:187 | evaluate_round 6 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:45:27,294 | server.py:223 | fit_round 7: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:45:27,325 | server.py:237 | fit_round 7 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:45:27,337 | server.py:125 | fit progress: (7, 0.46986773887557215, {'accuracy': 0.7666666666666667}, 0.6102112000000002)
DEBUG flwr 2023-09-18 17:45:27,337 | server.py:173 | evaluate_round 7: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:45:27,360 | server.py:187 | evaluate_round 7 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:45:27,360 | server.py:223 | fit_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:45:27,393 | server.py:237 | fit_round 8 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:45:27,408 | server.py:125 | fit progress: (8, 0.46986278317566105, {'accuracy': 0.7666666666666667}, 0.6806938000000002)
DEBUG flwr 2023-09-18 17:45:27,408 | server.py:173 | evaluate_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:45:27,435 | server.py:187 | evaluate_round 8 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:45:27,435 | server.py:223 | fit_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:45:27,474 | server.py:237 | fit_round 9 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:45:27,487 | server.py:125 | fit progress: (9, 0.46986274839528513, {'accuracy': 0.7666666666666667}, 0.7593086000000007)
DEBUG flwr 2023-09-18 17:45:27,487 | server.py:173 | evaluate_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:45:27,511 | server.py:187 | evaluate_round 9 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:45:27,511 | server.py:223 | fit_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:45:27,549 | server.py:237 | fit_round 10 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:45:27,562 | server.py:125 | fit progress: (10, 0.4698627482714406, {'accuracy': 0.7666666666666667}, 0.8354008999999998)
DEBUG flwr 2023-09-18 17:45:27,563 | server.py:173 | evaluate_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:45:27,594 | server.py:187 | evaluate_round 10 received 6 results and 0 failures
INFO flwr 2023-09-18 17:45:27,594 | server.py:147 | FL finished in 0.8669811000000003
INFO flwr 2023-09-18 17:45:27,594 | app.py:218 | app_fit: losses_distributed [(1, 0.4041576494773229), (2, 0.40221145848433176), (3, 0.4022129168113073), (4, 0.409671188110397), (5, 0.4094172545841762), (6, 0.4081718593835831), (7, 0.40781911739281246), (8, 0.4078189913715635), (9, 0.40781898753983636), (10, 0.40781898753983636)]
INFO flwr 2023-09-18 17:45:27,594 | app.py:219 | app_fit: metrics_distributed_fit {}
INFO flwr 2023-09-18 17:45:27,595 | app.py:220 | app_fit: metrics_distributed {}
INFO flwr 2023-09-18 17:45:27,595 | app.py:221 | app_fit: losses_centralized [(0, 0.6931471805599453), (1, 0.4772083827409828), (2, 0.47058630843542043), (3, 0.47057446153972515), (4, 0.4705744972100784), (5, 0.46989038504356073), (6, 0.46989277613899866), (7, 0.46986773887557215), (8, 0.46986278317566105), (9, 0.46986274839528513), (10, 0.4698627482714406)]
INFO flwr 2023-09-18 17:45:27,595 | app.py:222 | app_fit: metrics_centralized {'accuracy': [(0, 0.44333333333333336), (1, 0.7733333333333333), (2, 0.77), (3, 0.77), (4, 0.77), (5, 0.7733333333333333), (6, 0.7733333333333333), (7, 0.7666666666666667), (8, 0.7666666666666667), (9, 0.7666666666666667), (10, 0.7666666666666667)]}
>>>>>>>>>>>>> setting initial parameters
[[0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0.]]
>>>>>>>>>>>>> evaluating on server-side
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.99898007,  1.07426718,  0.83292752,  0.25135616,  0.28211267,
         0.0449611 ,  0.09113613,  0.00976405]]), array([-0.1545019])]
[array([[-1.99898007,  1.07426718,  0.83292752,  0.25135616,  0.28211267,
         0.0449611 ,  0.09113613,  0.00976405]]), array([-0.1545019])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.04515985,  1.00712234,  0.76906463,  0.30541321,  0.1907492 ,
         0.09815857,  0.02931062,  0.07151048]]), array([0.01777965])]
[array([[-2.04515985,  1.00712234,  0.76906463,  0.30541321,  0.1907492 ,
         0.09815857,  0.02931062,  0.07151048]]), array([0.01777965])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.04461417,  1.00679934,  0.76886531,  0.30525572,  0.19062471,
         0.09812583,  0.02937131,  0.0714836 ]]), array([0.01790442])]
[array([[-2.04461417,  1.00679934,  0.76886531,  0.30525572,  0.19062471,
         0.09812583,  0.02937131,  0.0714836 ]]), array([0.01790442])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.04461707,  1.00680182,  0.76886478,  0.30525623,  0.19062391,
         0.098127  ,  0.02937256,  0.0714829 ]]), array([0.01790481])]
[array([[-2.04461707,  1.00680182,  0.76886478,  0.30525623,  0.19062391,
         0.098127  ,  0.02937256,  0.0714829 ]]), array([0.01790481])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.98677132,  1.00009553,  0.77658734,  0.30759285,  0.1883974 ,
         0.0992176 ,  0.06004535,  0.05279197]]), array([-0.03060891])]
[array([[-1.98677132,  1.00009553,  0.77658734,  0.30759285,  0.1883974 ,
         0.0992176 ,  0.06004535,  0.05279197]]), array([-0.03060891])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.98693395,  1.00032608,  0.77664043,  0.30757629,  0.18840616,
         0.09923754,  0.05998296,  0.05274544]]), array([-0.03059494])]
[array([[-1.98693395,  1.00032608,  0.77664043,  0.30757629,  0.18840616,
         0.09923754,  0.05998296,  0.05274544]]), array([-0.03059494])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.9944598 ,  1.01566012,  0.8167565 ,  0.32829514,  0.18414044,
         0.08868176,  0.0268902 ,  0.0700381 ]]), array([-0.03912862])]
[array([[-1.9944598 ,  1.01566012,  0.8167565 ,  0.32829514,  0.18414044,
         0.08868176,  0.0268902 ,  0.0700381 ]]), array([-0.03912862])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.99453123,  1.0155906 ,  0.81678495,  0.3284316 ,  0.18407559,
         0.08878302,  0.02689843,  0.07004157]]), array([-0.03920618])]
[array([[-1.99453123,  1.0155906 ,  0.81678495,  0.3284316 ,  0.18407559,
         0.08878302,  0.02689843,  0.07004157]]), array([-0.03920618])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.99453136,  1.01559054,  0.81678512,  0.32843213,  0.18407465,
         0.08878364,  0.02689873,  0.07004144]]), array([-0.0392065])]
[array([[-1.99453136,  1.01559054,  0.81678512,  0.32843213,  0.18407465,
         0.08878364,  0.02689873,  0.07004144]]), array([-0.0392065])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.99453136,  1.01559054,  0.81678512,  0.32843213,  0.18407465,
         0.08878365,  0.02689873,  0.07004144]]), array([-0.0392065])]
[array([[-1.99453136,  1.01559054,  0.81678512,  0.32843213,  0.18407465,
         0.08878365,  0.02689873,  0.07004144]]), array([-0.0392065])]
>>>>>>>>>>>>>run time:
0.578125s
